Add steps to check and the solution to make KEDA work in proxy network

mashrin opened the pull request:
Solution for the issues like: https://github.com/kedacore/keda/issues/587

Will help any developers facing issues while installing KEDA in a proxy network.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/kedacore/keda-docs?pullRequest=83) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/kedacore/keda-docs?pullRequest=83) before we can accept your contribution.<br/><br/>:x: mashrin [sign now](https://cla.opensource.microsoft.com/kedacore/keda-docs?pullRequest=83)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/kedacore/keda-docs?pullRequest=83) it.</sub>

tomkerkhove commented:
What do you think @jeffhollan - Ok or do we introduce something like https://keda.sh/troubleshooting?

tomkerkhove commented on content/FAQ.md:
```diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+Check the API service status and the pods now. Should work!
+
+
```
Nitpick - 2 new lines, let's remove one
```suggestion
```

tomkerkhove commented on content/FAQ.md:
```diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
 
+#### How to make KEDA work in a proxy network?
+If while setting up KEDA, you get an error: "(v1beta1.external.metrics.k8s.io) status FailedDiscoveryCheck" with a Message: no response from https://ip:443: Get https://ip:443: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers); one of the reason for this can be that you are behind a proxy network.
+
```
Nitpick - 2 new lines, let's remove one
```suggestion
```

tomkerkhove commented on content/FAQ.md:
````diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+Find the apiservice name for the service keda/keda-metrics-apiserver:
+
+```kubectl get apiservice --all-namespaces```
````
When using "```" you should use multiple lines, for example:

> \`\`\`
> kubectl get apiservice --all-namespaces
> \`\`\`

Can you change this for all commands please?

mashrin replied:
Done

tomkerkhove commented on content/FAQ.md:
```diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+###### Check the status:
+
+Find the apiservice name for the service keda/keda-metrics-apiserver:
```
```suggestion
Find the api service name for the service `keda/keda-metrics-apiserver`:
```

tomkerkhove commented on content/FAQ.md:
````diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+```kubectl get apiservice --all-namespaces```
+
+Check for the status of the apiservice found in previous step:
````
```suggestion
Check for the status of the api service found in previous step:
```

tomkerkhove commented on content/FAQ.md:
````diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+Example: ```kubectl get apiservice v1beta1.external.metrics.k8s.io -o yaml```
+
+If the status is False, then there seems to be an issue and proxy network might be the primary reason for it.
````
```suggestion
If the status is `False`, then there seems to be an issue and proxy network might be the primary reason for it.
```

tomkerkhove commented on content/FAQ.md:
```diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+###### Follow these steps:
+
+Find the cluster IP for the keda-metrics-apiserver and keda-operator-metrics:
```
```suggestion
Find the cluster IP for the `keda-metrics-apiserver` and `keda-operator-metrics`:
```

tomkerkhove commented on content/FAQ.md:
````diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
+
+```kubectl get services --all-namespaces```
+
+In the ```/etc/kubernetes/manifests/kube-apiserver.yaml``` - add the cluster IPs found in the previous step in no_proxy variable.
````
```suggestion
In the `/etc/kubernetes/manifests/kube-apiserver.yaml` - add the cluster IPs found in the previous step in no_proxy variable.
```

tomkerkhove commented on content/FAQ.md:
```diff
@@ -46,6 +46,43 @@ There are several ways to get involved.
 Yes! KEDA is now 1.0 and suited for production workloads.
 
+#### How to make KEDA work in a proxy network?
+If while setting up KEDA, you get an error: "(v1beta1.external.metrics.k8s.io) status FailedDiscoveryCheck" with a Message: no response from https://ip:443: Get https://ip:443: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers); one of the reason for this can be that you are behind a proxy network.
```
```suggestion
If while setting up KEDA, you get an error: `(v1beta1.external.metrics.k8s.io) status FailedDiscoveryCheck` with a message: `no response from https://ip:443: Get https://ip:443: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)`.

 One of the reason for this can be that you are behind a proxy network.
```

mashrin commented:
@tomkerkhove Thanks for the review. Addressed all review comments.

Can we bold the text in the rendered page for these two lines:
(Line 54) ###### Check the status:
(Line 75)  ###### Follow these steps:

I have done it in the markdown but it's not rendered on the page.

anirudhgarg commented:
@tomkerkhove I am commenting on your idea of having a page for troubleshooting - I think that is a great idea.

tomkerkhove commented:
> @tomkerkhove Thanks for the review. Addressed all review comments.
> 
> Can we bold the text in the rendered page for these two lines:
> (Line 54) ###### Check the status:
> (Line 75) ###### Follow these steps:
> 
> I have done it in the markdown but it's not rendered on the page.

It looks bold in the preview, thanks!

tomkerkhove commented:
> @tomkerkhove I am commenting on your idea of having a page for troubleshooting - I think that is a great idea.

@anirudhgarg  I'll send a PR later today!
